ibc-core/ics24-host/types/src/error.rs

```diff
@@ -9,6 +9,35 @@ use ibc_primitives::prelude::*;
 use ibc_primitives::proto::Error as ProtoError;
 use prost::DecodeError as ProstError;
 
+/// Errors that originate from host implementations.
+#[derive(Debug, Display)]
+pub enum HostError {
+    /// application module error: `{description}`
+    AppModule { description: String },
+    /// invalid data: `{description}`
+    InvalidData { description: String },
+    /// missing data: `{description}`
+    MissingData { description: String },
```

Comment:
Does it make sense be naming these variants `InvalidState` and `MissingState`. Suggesting as these two are gonna be used in places we mostly do storage call and looking for a state.

Reply:
I think no matter which one we go with, it won't be semantically appropriate in every situation, unless we split it into two variants: `InvalidData` and `InvalidState`. But that would then open the door to getting more granular with every variant in `HostError`.

Reply:
[126f874](https://github.com/cosmos/ibc-rs/pull/1331/commits/126f87432dd3287c5a341bc22ae3bc52cab289a8)